src/importexport/mei/internal/meiimporter.cpp

```diff
@@ -1254,6 +1254,16 @@ bool MeiImporter::readSectionElements(pugi::xml_node parentNode)
 {
     bool success = true;
 
+    libmei::Section meiSection;
+    meiSection.Read(parentNode);
+
+    if (!m_score->measures()->empty() && meiSection.HasRestart() && meiSection.GetRestart() == libmei::BOOLEAN_true) {
+        Measure* measure = dynamic_cast<Measure*>(m_score->measures()->last());
+        if (measure) {
```

Comment:
The following would be more MuseScore-idiomatic:
```suggestion
        MeasureBase* mb = m_score->measures()->last();
        if (mb->isMeasure()) {
```
Alternatively, at the start of the method do:
```
MeasureBase* lastMeasureBase = !m_score->measures()->empty() ? m_score->measures()->last() : nullptr
```
and then check `if (lastMeasureBase)` or `lastMeasureBase && lastMeasureBase->isMeasure()` where appropriate (also at the bottom of the method)

Reply:
@cbjeukendrup Thanks, looks more elegant now. But unit test fail. Any idea?

Reply:
Ah, I think I see the potential problem: what's the last measure at the top of the method, might not be the last measure anymore at the bottom of the method. In between, `this->readMeasure` for example might add another measure.

So, it looks like you inevitably need to query the last measure twice.